src/com/cft/contactmerge/tests/CompareContactPartsTest.java

```diff
@@ -31,43 +30,648 @@ void doNamesMatch_Yes_IgnoreSpaces() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("Josephine Vitty", "Sa Democh", "Josephine-Mc-Vitty", "Sa-La-Democh"));
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("Joseph", "Lee", "Josephine-Mc-Vitty", "Sa-La-Cathleen"));
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("Josephine-Mc-Vitty", "Sa-La-Cathleen", "Joseph", "La Lee"));
+    }
+
+    @Test
+    void doNamesMatch_YesMaybeNo_FirstAndLastNamesContainMultiHyphens_CaseInsensitive() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("Josephine-mc-vitty", "Sa-la-democh", "vittY", "democH"));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("jOsephine-mc-vitty", "sA-La-democh", "mc vitTy", "la demoCh"));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("joSephine-mc-vitty", "sa-La-democh", "mC", "lA"));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("josEphine-mc-vitty", "sa-lA-democh", "josephine Mc", "sa La"));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("josePhine-mc-vitty", "sa-la-Democh", "josephinE", "sA"));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("josepHine-mc-vitty", "sa-la-dEmoch", "josephiNe vitty", "Sa democh"));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("Vitty", "Democh", "josephine-Mc-vitty", "sa-la-democH"));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("Mc vitty", "LA democh", "josephinE-mc-vitty", "sa-la-demoCh"));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("mC", "La", "josephiNe-mc-vitty", "sa-la-demOch"));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("josephine MC", "Sa la", "josephIne-mc-vitty", "sa-la-deMoch"));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("josephinE", "sA", "josepHine-mc-vitty", "sa-LA-democh"));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("josephine viTty", "sa dEmoch", "josePhine-mc-vitty", "Sa-la-democh"));
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("JosePh", "leE", "JosEphine-Mc-Vitty", "sa-la-Cathleen"));
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("JOsephine-Mc-Vitty", "sa-la-cathLeen", "joSeph", "la lEe"));
+    }
+
+    @Test
+    void doNamesMatch_YesMaybeNo_FirstAndLastNamesContainMultiHyphens_CaseInsensitive_IgnoreSpaces() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" Josephine-mc-vitty", "Sa-la-democh ", "vittY ", " democH"));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("jOsephine-mc-vitty ", " sA-La-democh", " mc vitTy", "la demoCh "));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch(" joSephine-mc-vitty ", " sa-La-democh ", "  mC  ", "  lA  "));
```

Comment:
You may already have it but this test reminds me that we should make sure that "Adams, Kathleen" does not match "Adamson, Lee" (i.e. that comparison should return "no").

Reply:
I think it's the same as what I have (("Joseph" vs "Josephine") and ("Cathleen" vs "Lee")) == no. Let me know if you think your pattern differs, and I will add it.

Reply:
Yep. Close anyways. My example had substring match on both the first and the last but if you do it in two separate tests, I think it is the same.